tests/handlers/test_cas.py

```diff
@@ -221,6 +221,33 @@ def test_map_cas_user_does_not_register_new_user(self) -> None:
         # check that the auth handler was not called as expected
         auth_handler.complete_sso_login.assert_not_called()
 
+    @override_config(
+        {"cas_config": {"allow_numeric_ids": True, "numeric_ids_prefix": "NUMERICUSER"}}
+    )
+    def test_register_user_with_numeric_ids_when_allowed(self) -> None:
+        """Ensures new users with numeric user IDs are registered if the allow_numeric_ids flag is enabled."""
+
+        # stub out the auth handler
+        auth_handler = self.hs.get_auth_handler()
+        auth_handler.complete_sso_login = AsyncMock()  # type: ignore[method-assign]
+
+        cas_response = CasResponse("test_user", {})
+        request = _mock_request()
+        self.get_success(
+            self.handler._handle_cas_response(request, cas_response, "redirect_uri", "")
+        )
+
+        # check that the auth handler got called as expected
+        auth_handler.complete_sso_login.assert_called_once_with(
+            "@1234:test",
```

Comment:
should this not  say 
```suggestion
            "@NUMERICUSER1234:test",
```

or have I missed the point?

Reply:
Yes, something's wrong, I'm not comfortable with devops :(
The aim of the test is to try to connect with a numeric ID and with the CAS parameters that allow it, it must work to pass the test. So I don't know how to change the code to "1234" instead of "@1234:test" which is supposed to be formed during the login process or account creation.

Reply:
I've removed the test part, that should be done by a real devops who knows synapse well.